src/libs/auth.js

```diff
@@ -19,29 +20,94 @@ import {
 import * as Utils from 'src/libs/utils'
 
 
+export const getOidcConfig = () => {
+  const metadata = {
+    authorization_endpoint: `${getConfig().orchestrationUrlRoot}/oauth2/authorize`,
+    token_endpoint: `${getConfig().orchestrationUrlRoot}/oauth2/token`,
+    ...(isGoogleAuthority() && {
+      userinfo_endpoint: 'https://openidconnect.googleapis.com/v1/userinfo',
+      revocation_endpoint: 'https://oauth2.googleapis.com/revoke'
+    })
+  }
+  return {
+    authority: `${getConfig().orchestrationUrlRoot}/oauth2/authorize`,
+    client_id: authStore.get().oidcConfig.clientId,
+    popup_redirect_uri: `${window.origin}/redirect-from-oauth`,
+    silent_redirect_uri: `${window.origin}/redirect-from-oauth-silent`,
+    metadata,
+    prompt: 'consent login',
+    scope: 'openid email profile',
+    loadUserInfo: isGoogleAuthority(),
+    userStore: new WebStorageStateStore({ store: window.localStorage }),
+    automaticSilentRenew: true,
+    includeIdTokenInSilentRenew: true,
+    extraQueryParams: { access_type: 'offline' },
+    revokeTokenTypes: ['access_token', 'refresh_token']
+  }
+}
+
+const isGoogleAuthority = () => {
+  return _.startsWith('https://accounts.google.com', authStore.get().oidcConfig.authorityEndpoint)
+}
+
 const getAuthInstance = () => {
-  return window.gapi.auth2.getAuthInstance()
+  return authStore.get().authContext
 }
 
 export const signOut = () => {
   // Note: Ideally, we'd actually clear the Leo proxy cookie here, but there's not currently an endpoint to do that.
   // When IA-2236 is done, add that call here.
   cookieReadyStore.reset()
   sessionStorage.clear()
-  getAuthInstance().signOut()
+  const auth = getAuthInstance()
+  auth.revokeTokens()
+    .catch(() => false)
+    .then(() => auth.removeUser())
+    .then(() => auth.clearStaleState())
```

Comment:
Would it be desirable to call the subsequent functions when a previous fails, like so?

```suggestion
    .finally(() => auth.removeUser())
    .finally(() => auth.clearStaleState())
```

Reply:
`finally` does seem nicer, and works as intended. Will change.

Reply:
I also wonder if we need to do any reporting/alerting in case one or more of those three calls fail, for security purposes.

Cc: @dinvlad

Reply:
`revokeTokens` calls the token revoke endpoint and _can_ fail if the token is already revoked. So I think I'm going to leave the `catch` there to recover errors. `removeUser` and `clearStaleState` just modifies browser state, so makes sense to do them in finally (we want them to always occur).

Reply:
I see. 

> revokeTokens calls the token revoke endpoint and can fail if the token is already revoked.

Would it make sense / be possible to recover from that specific error and not the others in case we'd not want to swallow other error modes? I don't know how fussy we want to be about this security-wise 😊

Reply:
I've noticed in the wild that it's actually expected that it will fail (there's a race condition in there).  In the case of google it revokes 2 tokens: the refresh_token and the access_token.  If you revoke one, it seems to revoke the other but there's some time t where that hasn't happened yet (< a second it seems)

Reply:
Makes sense, just pushed a change to only recover from `invalid_token` errors. :)